finality-provider/service/fp_instance.go

```diff
@@ -246,53 +186,64 @@ func (fp *FinalityProviderInstance) finalitySigSubmissionLoop() {
 				continue
 			}
 			fp.logger.Info(
-				"successfully submitted a finality signature to the consumer chain",
+				"successfully submitted the finality signature to the consumer chain",
+				zap.String("consumer_id", string(fp.GetChainID())),
 				zap.String("pk", fp.GetBtcPkHex()),
-				zap.Uint64("height", b.Height),
+				zap.Uint64("start_height", pollerBlocks[0].Height),
+				zap.Uint64("end_height", targetHeight),
 				zap.String("tx_hash", res.TxHash),
 			)
 
-		case targetBlock := <-fp.laggingTargetChan:
-			res, err := fp.tryFastSync(targetBlock)
-			fp.isLagging.Store(false)
+		case <-fp.quit:
+			fp.logger.Info("the finality signature submission loop is closing")
+			return
+		}
+	}
+}
+
+func (fp *FinalityProviderInstance) getAllBlocksFromChan() []*types.BlockInfo {
+	var pollerBlocks []*types.BlockInfo
+	for {
+		select {
+		case b := <-fp.poller.GetBlockInfoChan():
+			shouldProcess, err := fp.shouldProcessBlock(b)
```

Comment:
there might be a problem here since `shouldProcessBlock` makes RPC calls to the Babylon chain. so in this `for` loop we are doing a lot of RPC calls.

Reply:
Agreed this case could happen in case of catching up. Left a TODO here for now